pkg/processor/runtime/nodejs/js/wrapper.js

```diff
@@ -0,0 +1,265 @@
+/*
+Copyright 2017 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+const net = require('net')
+const events = require('events')
+
+const jsonCtype = 'application/json'
+const initContextFunctionName = 'initContext'
+
+let context = {
+    userData: {},
+    callback: async (handlerResponse) => {
+        context._eventEmitter.emit('callback', handlerResponse)
+    },
+    Response: Response,
+    logger: {
+        error: (message, withData) => log('error', message, withData),
+        warn: (message, withData) => log('warning', message, withData),
+        info: (message, withData) => log('info', message, withData),
+        debug: (message, withData) => log('debug', message, withData),
+        errorWith: (message, withData) => log('error', message, withData),
+        warnWith: (message, withData) => log('warning', message, withData),
+        infoWith: (message, withData) => log('info', message, withData),
+        debugWith: (message, withData) => log('debug', message, withData),
+    },
+    _socket: undefined,
+    _eventEmitter: new events.EventEmitter(),
+}
+
+function Response(body = null,
+                  headers = null,
+                  contentType = 'text/plain',
+                  statusCode = 200) {
+    this.body = body
+    this.headers = headers
+    this.content_type = contentType
+    this.status_code = statusCode
+
+    if (!isString(this.body)) {
+        this.body = JSON.stringify(this.body)
+        this.content_type = jsonCtype
+    }
+}
+
+function log(level, message, withData) {
+    if (withData === undefined) {
+        withData = {}
+    }
```

Comment:
That is, im afraid, not supported on nodejs 10

```javascript
SyntaxError: Unexpected token ?
    at new Script (vm.js:83:7)
    at createScript (vm.js:277:10)
...
```

Reply:
Oh, right, I missed this one.
Well you can still use the older `||` operator:
```suggestion
    const data = withData || {}
```
The only difference is that it will default to `{}` not only for `null` and `undefined` but to all other “falsy” values: `false`, `NaN`, `0`, `-0`, `''`. If any of them are valid for `withData`, i.e. if having `withData` equals one of them you _don't_ want it to default to `{}` — then you can't use `||`. Otherwise, go ahead.

---

Now when I think about it, if you only want to check if `withData` param is provided or not, and default to `{}` in case it was not — you can simply use default params:
```js
function log(level, message, withData = {}) {
```
🤦

Reply:
such a hard decision to pass an arg with a  default value 😆